sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -755,6 +756,81 @@ def create_builtin_provider():
                           no_input_transforms=('Create', ))
 
 
+class TranslatingProvider(Provider):
+  def __init__(
+      self,
+      transforms: Mapping[str, Callable[..., beam.PTransform]],
+      underlying_provider: Provider):
+    self._transforms = transforms
+    self._underlying_provider = underlying_provider
+
+  def provided_transforms(self):
+    return self._transforms.keys()
+
+  def available(self):
+    return self._underlying_provider.available()
+
+  def cache_artifacts(self):
+    return self._underlying_provider.cache_artifacts()
+
+  def underlying_provider(self):
+    return self._underlying_provider
+
+  def to_json(self):
+    return {'type': "TranslatingProvider"}
+
+  def create_transform(
+      self, typ: str, config: Mapping[str, Any],
+      yaml_create_transform: Any) -> beam.PTransform:
+    return self._transforms[typ](self._underlying_provider, **config)
```

Comment:
Does there need to be a new provider? Seems as though `RenamingProvider` has all the relevant functionality and would allow mappings and defaults should there be Java built-in transforms that expose those in the future

Reply:
This does a deeper rewriting of the arguments than renaming, though I was thinking that the renaming transform could be defined as a special case of this one. (The SQL-based one possibly as well.)

Reply:
Yeah I think they could be merged in either direction, but not a blocker. LGTM